library/Requests/Transport/cURL.php

```diff
@@ -219,8 +219,20 @@ public function request_multiple($requests, $options) {
 
 			do {
 				$status = curl_multi_exec($multihandle, $active);
+				// Return immediately if there's work to be done.
+				if ($status !== CURLM_CALL_MULTI_PERFORM) {
+					break;
+				}
+				// Block until there is activity on any of the curl_multi connections.
+				// Timeouts can be specified in microseconds.
+				// Provide curl_multi_select with a timeout of 100ms (May not be necessary).
+				// As a fallback, if curl_multi_select couldn't properly asynchronously wait for input, sleep for 0.1 ms.
+				$select_status = curl_multi_select($multihandle, 0.1);
+				if ($select_status === -1) {
+					usleep(100);
+				}
 			}
-			while ($status === CURLM_CALL_MULTI_PERFORM);
+			while (true);
```

Comment:
Actually, the above approach may be incorrect. Maybe the part that was causing the busy loop wasn't the above `do{}while`, but below 

The below is probably a better alternative, and can get rid of the above.

```
                       while ($status === CURLM_CALL_MULTI_PERFORM);
 
+                       // Insert a 50ms sleep here so that we don't busy loop and chew cpu
+                       $selectRes  = curl_multi_select($multihandle, 0.05);
+
                        $to_process = array();
```